protocol/contracts/templegold/TempleGold.sol

```diff
@@ -0,0 +1,387 @@
+    }
+
+    /**
+     * @notice Set dai gold auction contract address
+     * @param _daiGoldAuction  contract address
+     */
+    function setDaiGoldAuction(address _daiGoldAuction) external override onlyOwner onlySourceChain {
+        if (_daiGoldAuction == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        daiGoldAuction = IDaiGoldAuction(_daiGoldAuction);
+        emit DaiGoldAuctionSet(_daiGoldAuction);
+    }
+
+    /**
+     * @notice Set team gnosis address
+     * @param _gnosis Team gnosis address
+     */
+    function setTeamGnosis(address _gnosis) external override onlyOwner onlySourceChain {
+        if (_gnosis == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        teamGnosis = _gnosis;
+        emit TeamGnosisSet(_gnosis);
+    }
+
+    /**
+     * @notice Whitelist an address to allow transfer of Temple Gold to or from
+     * @param _contract Contract address to whitelist
+     * @param _whitelist Boolean whitelist state
+     */
+    function authorizeContract(address _contract, bool _whitelist) external override onlyOwner {
+        if (_contract == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        authorized[_contract] = _whitelist;
+        emit ContractAuthorizationSet(_contract, _whitelist);
+    } 
+
+    /**
+     * @notice Set distribution percentages of newly minted Temple Gold
+     * @param _params Distribution parameters
+     */
+    function setDistributionParams(DistributionParams calldata _params) external override onlyOwner onlySourceChain {
+        if (_params.staking + _params.gnosis + _params.daiGoldAuction != DISTRIBUTION_DIVISOR) { revert ITempleGold.InvalidTotalShare(); }
+        distributionParams = _params;
+        emit DistributionParamsSet(_params.staking, _params.daiGoldAuction, _params.gnosis);
+    }
+
+    /**
+     * @notice Set vesting factor
+     * @param _factor Vesting factor
+     */
+    function setVestingFactor(VestingFactor calldata _factor) external override onlyOwner onlySourceChain {
+        if (_factor.value == 0 || _factor.weekMultiplier == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        /// @dev initialize
+        if (lastMintTimestamp == 0) { lastMintTimestamp = uint32(block.timestamp); }
+        else { mint(); }
+        vestingFactor = _factor;
+        emit VestingFactorSet(_factor.value, _factor.weekMultiplier);
+    }
+    
+    /**
+     * @notice Mint new tokens to be distributed. Open to call from any address
+     * Enforces minimum mint amount and uses vesting factor to calculate mint token amount.
+     * Minting is only possible on source chain Arbitrum
+     */
+    function mint() public override onlySourceChain {
+        VestingFactor memory vestingFactorCache = vestingFactor;
+        DistributionParams storage distributionParamsCache = distributionParams;
+        if (vestingFactorCache.value == 0) { revert ITempleGold.MissingParameter(); }
```

Comment:
Wouldn't it be better to NOT revert here and simply silently return?
Functions like `DaiGoldAuction.startAuction()` rely on `TempleGold.mint()`, so if someone sets `vestingFactorCache.value == 0` for any reason, then you have the auctions blocked. 

And a silent return here would not hurt. (as you already do in the lines below).

Reply:
The idea is for the first transactions after deployment before setting vesting factor.
A rare case that vesting factor would be reset to 0 after.  I will use a silent return

Reply:
got it, that makes sense as a prevention before it is initialized.